crates/red_knot_project/src/watch/project_watcher.rs

```diff
@@ -68,41 +68,47 @@ impl ProjectWatcher {
 
         self.has_errored_paths = false;
 
-        let project_path = db
-            .system()
-            .canonicalize_path(&project_path)
-            .unwrap_or(project_path);
-
         let config_paths = db
             .project()
             .metadata(db)
             .extra_configuration_paths()
             .iter()
-            .cloned();
+            .map(SystemPathBuf::as_path);
+
+        // Find the common ancestor between the project root and any path provided by the user on the CLI.
```

Comment:
I don't think `deduplicate_nested_paths` really find the common ancestor? If you give it two unrelated paths it will return both of them, not their common ancestor. It just removes paths that are entirely nested within another one. 
```suggestion
        // Watch both the project root and any paths provided by the user on the CLI (removing any redundant nested paths).
```